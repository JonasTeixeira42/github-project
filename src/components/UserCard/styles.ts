import styled, { css } from 'styled-components'

export const Wrapper = styled.article`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    flex-direction: column;
    padding: ${theme.spacings.medium};
    box-shadow: ${theme.shadow.default};
    border-radius: ${theme.border.radius};
    background: ${theme.colors.white};
  `}
`

export const Image = styled.img`
  height: 15rem;
  width: 15rem;
  border-radius: 50%;
  object-fit: cover;

  background: #f6f7f8;
  background-image: linear-gradient(
    to right,
    #f6f7f8 0%,
    #edeef1 20%,
    #f6f7f8 40%,
    #f6f7f8 100%
  );
  background-size: 80rem 14rem;
  animation: placeholderShimmer 1s linear infinite forwards;

  @keyframes placeholderShimmer {
    0% {
      background-position: -40rem 0;
    }

    100% {
      background-position: 40rem 0;
    }
  }
`

export const Content = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.small};
    text-align: center;
  `}
`

export const Id = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.gray};
  `}
`

export const Name = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`
