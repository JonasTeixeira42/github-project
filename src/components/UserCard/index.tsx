import React from 'react'

import * as S from './styles'

export type UserCardProps = {
  id?: string
  login?: string
  avatar_url?: string
}

const UserCard = ({ id = '', login = '', avatar_url = '' }: UserCardProps) => {
  return (
    <S.Wrapper>
      <S.Image src={avatar_url} alt={login} />
      <S.Content>
        <S.Name>{login}</S.Name>
        <S.Id>{id}</S.Id>
      </S.Content>
    </S.Wrapper>
  )
}

export default UserCard
