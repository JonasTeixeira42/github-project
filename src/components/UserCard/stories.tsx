import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import UserCard, { UserCardProps } from '.'

export default {
  title: 'UserCard',
  component: UserCard,
  args: {
    image: 'https://source.unsplash.com/random/100x100',
    name: 'JonasTeixeira42',
    id: '1'
  }
} as Meta

export const Default: Story<UserCardProps> = (args) => (
  <div style={{ maxWidth: '30rem', margin: '0 auto' }}>
    <UserCard {...args} />
  </div>
)
