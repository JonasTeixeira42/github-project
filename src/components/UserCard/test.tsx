import React from 'react'
import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import UserCard from '.'

const props = {
  avatar_url: 'https://source.unsplash.com/random/100x100',
  login: 'JonasTeixeira42',
  id: 1
}

describe('<UserCard />', () => {
  it('should render the UserCard component', () => {
    const { container } = renderWithTheme(<UserCard {...props} />)

    expect(
      screen.getByRole('heading', { name: /JonasTeixeira42/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('heading', { name: /1/i })).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: /JonasTeixeira42/i })
    ).toHaveAttribute('src', 'https://source.unsplash.com/random/100x100')

    expect(container.firstChild).toMatchSnapshot()
  })
})
