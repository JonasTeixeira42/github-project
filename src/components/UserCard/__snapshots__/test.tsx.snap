// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> should render the UserCard component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 3.2rem;
  box-shadow: 0px 3px 6px #00000029;
  border-radius: 0.4rem;
  background: #ffffff;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
}

.c0:hover {
  -webkit-transform: translateY(-1rem);
  -ms-transform: translateY(-1rem);
  transform: translateY(-1rem);
}

.c1 {
  height: 15rem;
  width: 15rem;
  border-radius: 50%;
  object-fit: cover;
  background: #f6f7f8;
  background-image: linear-gradient( to right, #f6f7f8 0%, #edeef1 20%, #f6f7f8 40%, #f6f7f8 100% );
  background-size: 80rem 14rem;
  -webkit-animation: placeholderShimmer 1s linear infinite forwards;
  animation: placeholderShimmer 1s linear infinite forwards;
}

.c2 {
  margin-top: 2.4rem;
  text-align: center;
}

.c4 {
  font-size: 1.8rem;
  color: #808080;
}

.c3 {
  font-size: 1.8rem;
}

<article
  class="c0"
>
  <img
    alt="JonasTeixeira42"
    class="c1"
    src="https://source.unsplash.com/random/100x100"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      JonasTeixeira42
    </h2>
    <h3
      class="c4"
    >
      1
    </h3>
  </div>
</article>
`;
